test/libsolidity/util/TestFileParserTests.cpp

```diff
@@ -958,6 +959,43 @@ BOOST_AUTO_TEST_CASE(library)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(call_effects)
+{
+	std::map<std::string, Builtin> builtins;
+	builtins["builtin_returning_call_effect"] = [](FunctionCall const&) -> std::optional<bytes>
+	{
+	  return util::toBigEndian(u256(0x1234));
+	};
+	char const* source = R"(
+		// builtin_returning_call_effect -> 1
+		// - bla
+		// - bla bla
+		// - bla bla bla
+	)";
+	vector<FunctionCall> calls = parse(source, builtins);
+	BOOST_REQUIRE_EQUAL(calls.size(), 1);
+	BOOST_REQUIRE_EQUAL(calls[0].effects.size(), 3);
+	BOOST_REQUIRE_EQUAL(boost::trim_copy(calls[0].effects[0]), "bla");
+	BOOST_REQUIRE_EQUAL(boost::trim_copy(calls[0].effects[1]), "bla bla");
+	BOOST_REQUIRE_EQUAL(boost::trim_copy(calls[0].effects[2]), "bla bla bla");
+	source = R"(
+		// builtin_returning_call_effect -> 1
+		// - bla
+		// - bla bla
+		// builtin_returning_call_effect -> 2
+		// - bla bla bla
+		// builtin_returning_call_effect -> 3
+	)";
```

Comment:
Can you add more corner cases?
- Empty effects (i.e. just a dash). One or more.
- `// - - - `
- `// - > effect`
- `// - abc - def - ghi` - will this be one effect or three?

Reply:
Let's please not overdo testing of testing tools.